src/lib/crypto/kimchi_backend/gadgets/xor.ml

```diff
@@ -0,0 +1,333 @@
+open Core_kernel
+
+open Kimchi_backend_common.Plonk_constraint_system.Plonk_constraint
+
+module Bignum_bigint = Snarky_backendless.Backend_extended.Bignum_bigint
+
+(* XOR *)
+
+(* Boolean Xor of length bits *)
+let bxor (type f)
+    (module Circuit : Snarky_backendless.Snark_intf.Run with type field = f)
+    (input1 : Circuit.Field.t) (input2 : Circuit.Field.t) (length : int)
+    (len_xor : int) : Circuit.Field.t =
```

Comment:
I think you want to assert `length > 0` early on here.

Reply:
I think you also want to assert `len_xor > 0` or use uint.

Reply:
Assertions would be nice. There are no unsigned integers in standard libraries in OCaml. If you need one, use the `integers` package (https://github.com/yallop/ocaml-integers), which is already used in a number of sub-libraries in Mina

Reply:
asserts will suffice then